newIDE/app/src/MainFrame/index.js

```diff
@@ -214,35 +214,49 @@ const MainFrame = (props: Props) => {
     console.log(state);
   });
 
-  React.useEffect(() => {
-    (async () => {
-      if (!props.integratedEditor) openStartPage();
-      GD_STARTUP_TIMES.push(['MainFrameComponentDidMount', performance.now()]);
-      const { initialFileMetadataToOpen } = props;
-      try {
-        await _loadExtensions();
-        // Enable the GDJS development watcher *after* the extensions are loaded,
-        // to avoid the watcher interfering with the extension loading (by updating GDJS,
-        // which could lead in the extension loading failing for some extensions as file
-        // are removed/copied).
-        await setState(state => ({
-          ...state,
-          gdjsDevelopmentWatcherEnabled: true,
-        }));
-        if (initialFileMetadataToOpen) {
-          _openInitialFileMetadata(/* isAfterUserInteraction= */ false);
-        } else if (props.introDialog && !Window.isDev()) _openIntroDialog(true);
+  const _openIntroDialog = React.useCallback(
+    (open: boolean = true) => {
+      setState(state => ({ ...state, introDialogOpen: open }));
+    },
+    [setState]
+  );
 
+  const { integratedEditor, initialFileMetadataToOpen, introDialog } = props;
+  React.useEffect(
+    () => {
+      (() => {
+        if (!integratedEditor) openStartPage();
         GD_STARTUP_TIMES.push([
-          'MainFrameComponentDidMountFinished',
+          'MainFrameComponentDidMount',
           performance.now(),
         ]);
-        console.info('Startup times:', getStartupTimesSummary());
-      } catch {
-        /* Ignore errors */
-      }
-    })();
-  }, []);
+        _loadExtensions().then(() => 
+          // Enable the GDJS development watcher *after* the extensions are loaded,
+          // to avoid the watcher interfering with the extension loading (by updating GDJS,
+          // which could lead in the extension loading failing for some extensions as file
+          // are removed/copied).
+          setState(state => ({
+            ...state,
+            gdjsDevelopmentWatcherEnabled: true,
+          }))
+        ).then(state => {
+          if (initialFileMetadataToOpen) {
+            _openInitialFileMetadata(/* isAfterUserInteraction= */ false);
+          } else if (introDialog && !Window.isDev()) _openIntroDialog(true);
+
+          GD_STARTUP_TIMES.push([
+            'MainFrameComponentDidMountFinished',
+            performance.now(),
+          ]);
+          console.info('Startup times:', getStartupTimesSummary());
+        })
+        .catch(() => {
+          /* Ignore errors */
+        })
+      })();
+    },
+    [integratedEditor, initialFileMetadataToOpen, setState, _openIntroDialog]
```

Comment:
Linter warning asks to add `_openInitialFileMetadata`, `openStartPage`,`_loadExtensions` and `introDialog`.

I initially used `useCallback` for `_loadExtensions`, but including `_loadExtensions` (after useCallback) and `introDialog` into the deps list results in not opening of projects. I don't really understand why this happens :/

`_openInitialFileMetadata` and  `openStartPage` are basically rabbit holes, they depend upon a lot of other functions, which directly depend upon `state`. Wrapping all of them in a `useCallback` would mean that this `useEffect` will run on every render. Do you have some suggestions?